repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py

```diff
@@ -0,0 +1,296 @@
+        Find the target repository that is the best-match to the source one with the given
+        target PES ID.
+
+        :param src_pesidrepo: The source repository to find equivalent to.
+        :type src_pesidrepo: PESIDRepositoryEntry
+        :param target_pesid: The target PES ID which the target repository must contain.
+        :type target_pesid: str
+        :return: A target equivalent of given repository.
+        :rtype: Optional[PESIDRepositoryEntry]
+        """
+
+        candidates = []
+        for candidate in self.get_target_pesid_repos(target_pesid):
+            matches_rhui = candidate.rhui == src_pesidrepo.rhui
+            matches_repo_type = candidate.repo_type == 'rpm'
+            matches_arch = candidate.arch == api.current_actor().configuration.architecture
+
+            if matches_rhui and matches_arch and matches_repo_type:
+                # user can specify in future the specific channel should be
+                # prioritized always (e.g. want to go to EUS...).
+                channel = self.prio_channel or src_pesidrepo.channel
+                if candidate.channel == channel:
+                    return candidate
+                candidates.append(candidate)
+
+        # Fallback...
+        # Could not find exact-match, so go through candidates if we find an
+        # alternative in one of default channels (usually just 'ga')
+        for channel in self.default_channels:
+            for candidate in candidates:
+                if channel == candidate.channel:
+                    return candidate
+
+        # This is ...weird case (handling out of the method)
+        return None
+
+    def get_mapped_target_pesid_repos(self, src_pesidrepo):
+        """
+        Returns the dict of form {target_pesid: target PESIDRepositoryEntry} containing
+        pesids of repositories that is the source pesidrepo mapped to as keys and with
+        the best fitting repository for each of the target pesids as values.
+
+        The function always returns the best candidate for every target pesid
+        that fits to the given source repository. In case no repository is find
+        for a pesid, the value in dict is None: {pesid: None}
+
+        :param src_pesidrepo: The PESIDRepositoryEntry to find the target pesids and the corresponding
+                              best-fitting repositories to.
+        :type src_pesidrepo: PESIDRepositoryEntry
+        :return: A dictionary of the form described above.
+        :rtype: Dict[str, PESIDRepositoryEntry]
+        """
+        result = {}
+        for target_pesid in self.get_target_pesids(src_pesidrepo.pesid):
+            result[target_pesid] = self._find_repository_target_equivalent(src_pesidrepo, target_pesid)
+        return result
+
+    def get_mapped_target_repoids(self, src_pesidrepo):
+        """
+        Return the list of target repoids for the given src_pesidrepo.
+
+        Some actors do not has to check whether a target repository has been
```

Comment:
```suggestion
        Some actors do not have to check whether a target repository has been
```

Reply:
I would like to skip this one as I would have to fix this triple times in three actors and it's private docstring. But you are right otherwise. We have bunch of other typos in docstrings. Let's fix them during the refactoring - when it's possible we change the lib significantly..